services/QuillLMS/app/workers/clever_integration/hydrate_teacher_classrooms_cache_worker.rb

```diff
@@ -5,14 +5,13 @@ class HydrateTeacherClassroomsCacheWorker
     include Sidekiq::Worker
     sidekiq_options queue: SidekiqQueue::CRITICAL_EXTERNAL
 
-    def perform(teacher_id)
-      return unless clever_id?(teacher_id)
+    def perform(user_id)
+      user = ::User.find_by(id: user_id)
 
-      TeacherClassroomsCacheHydrator.run(teacher_id)
-    end
+      return if user.nil?
+      return unless user.teacher? && user.clever_authorized?
```

Comment:
This auth logic should be moved to TeachersController - auth happens in controllers by convention. In fact, `TeachersController#retrieve_classrooms`, where this worker is called, already contains auth conditionals.

Reply:
That's a good point.  I had a feeling this was redundant.